### ISSUE TITLE
Need guidance on using Glaze library with AUR/Arch API for JSON parsing

### ISSUE THREAD

**Issue by sm9cc** [stephenberry/glaze#431]
I am currently trying to use Glaze to parse JSON responses from the Arch/AUR API, but I'm having some difficulty understanding the process. 

Below is my code:

```c++
#include <cpr/cpr.h>
#include <fmt/core.h>
#include <glaze/glaze.hpp>

struct Result {
    std::string Description;
    int FirstSubmitted;
    int ID;
    int LastModified;
    std::string Maintainer;
    std::string Name;
    int NumVotes;
    std::string OutOfDate;
    std::string PackageBase;
    int PackageBaseID;
    double Popularity;
    std::string URL;
    std::string URLPath;
    std::string Version;
};

template <> struct glz::meta<Result> {
    using T                     = Result;
    static constexpr auto value = object("Description", &T::Description, "FirstSubmitted", &T::FirstSubmitted,
        "LastModified", &T::LastModified, "Maintainer", &T::Maintainer, "Name", &T::Name, "NumVotes", &T::NumVotes,
        "OutOfDate", &T::OutOfDate, "PackageBase", &T::PackageBase, "PackageBaseID", &T::PackageBaseID, "Popularity",
        &T::Popularity, "URL", &T::URL, "URLPath", &T::URLPath, "Version", &T::Version);
};

struct SearchResult {
    int resultcount;
    std::vector<Result> results;  // I don't think this is correct.
    std::string type;
    int version;
};

template <> struct glz::meta<SearchResult> {
    using T = SearchResult;
    static constexpr auto value
        = object("resultcount", &T::resultcount, "results", &T::results, "type", &T::type, "version", &T::version);
};

int main() {
    const auto response = cpr::Get(cpr::Url{"https://aur.archlinux.org/rpc?v=5&type=search&by=name&arg=glaze"});

    if (response.status_code != 200) {
        throw std::runtime_error("Invalid response (HTTP Status: " + std::to_string(response.status_code)
            + ", Error message:" + response.error.message + ")");
    }

    SearchResult searchResult;
    const auto error = glz::read_json(searchResult, response.text);

    if (error) {
        char buffer[1024];
        const auto e = format_error(error, buffer);

        fmt::print("Error: {}\n", e);
    }

    // The resultcount is correct.
    fmt::print("Result count: {}\n", searchResult.resultcount);

    // Type, Version, and results are not.
    fmt::print("Type: {}\n", searchResult.type);
    fmt::print("Version: {}\n", searchResult.version);

    for (const auto& result : searchResult.results) {
        fmt::print("Name: {}\n", result.Name);
    }

    return 0;
}

```

```Error: syntax_error
Result count: 2
Type: 
Version: 0
Name: 

Process finished with exit code 0
```

For reference, this is the API URL I'm working with: https://aur.archlinux.org/rpc?v=5&type=search&by=name&arg=glaze

Thank you in advance for your assistance.

**Comment by sm9cc** [comment]
I have done what you suggested: https://gist.github.com/sm9cc/ff5553991626058bf7e8243a890dc0e0#file-main-cpp

And now I get the following:

```terminate called after throwing an instance of 'std::runtime_error'
  what():  1:160: syntax_error
   {"resultcount":2,"results":[{"Description":"A High-Performance, In-Memory JSON and Interface Library for Modern C++","FirstSubmitted":1695013007,"ID":1321793,"LastModified":1695205181,"Maintainer":"sm9cc","Name":"glaze","NumVotes":0,"OutOfDate":null,"PackageBase":"glaze","PackageBaseID":197681,"Popularity":0,"URL":"https://github.com/stephenberry/glaze","URLPath":"/cgit/aur.git/snapshot/glaze.tar.gz","Version":"1.4.2-1"},{"Description":"A High-Performance, In-Memory JSON and Interface Library for Modern C++","FirstSubmitted":1695013571,"ID":1320524,"LastModified":1695013571,"Maintainer":"sm9cc","Name":"glaze-git","NumVotes":0,"OutOfDate":null,"PackageBase":"glaze-git","PackageBaseID":197682,"Popularity":0,"URL":"https://github.com/stephenberry/glaze","URLPath":"/cgit/aur.git/snapshot/glaze-git.tar.gz","Version":"v1.4.1.r14.da5dc36-1"}],"type":"search","version":5}
                                                                                                                                                                  ^


Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```

Thanks again.

**Comment by stephenberry** [comment]
Oh, I missed the fact that you also need to register "ID" with `Result`. It's missing.

**Comment by sm9cc** [comment]
> Oh, I missed the fact that you also need to register "ID" with `Result`. It's missing. Just add `"ID", &T::ID,`

Thank you for bringing this to my attention, It's working perfectly now!

> Again, this reveals an unhelpful error message, so I'll make an issue to improve the error messaging based on this test case.

Much appreciated.

Additionally, if you'd like to mention it in the README, I maintain two AUR packages: [glaze](https://aur.archlinux.org/packages/glaze) and [glaze-git](https://aur.archlinux.org/packages/glaze-git). As alternative ways for people to install and use this library on Arch & Arch based distributions.

**Comment by stephenberry** [comment]
I added a note about the AUR packages, thank you! And, I just merged in #433, which provides much better errors, especially for non-prettified JSON.
